Lib/fontTools/feaLib/parser.py

```diff
@@ -12,21 +12,33 @@
 
 
 class Parser(object):
+    """Initializes a Parser object.
+
+    Example:
+
+        .. code:: python
+
+            from fontTools.feaLib.parser import Parser
+            parser = Parser(file, font.getReverseGlyphMap())
+            parsetree = parser.parse()
+
+    Note: the ``glyphNames`` iterable serves a double role to help distinguish
+    glyph names from ranges in the presence of hyphens and to ensure that glyph
+    names referenced in a feature file are actually part of a font's glyph set.
+    If the iterable is left empty, no glyph name in glyph set checking takes
+    place.
```

Comment:
... and all glyph tokens containing hyphen are treated as literal glyph names, not as ranges. Adding space around the hyphen can also help disambiguate these.